lib/smack/SmackInstGenerator.cpp

```diff
@@ -640,11 +651,12 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst &ci) {
     auto mainFunction = cast<const Function>(castExpr);
     emit(Stmt::call(mainFunction->getName(), {}, {}));
 
-  } else if (name.find(Naming::RUST_PANIC1) != std::string::npos ||
-             name.find(Naming::RUST_PANIC2) != std::string::npos) {
+  } else if (is_rust_panic(name)) {
     // Convert Rust's panic functions into assertion violations
     emit(Stmt::assert_(Expr::lit(false),
                        {Attr::attr(Naming::RUST_PANIC_ANNOTATION)}));
+    // Block further assertions on this path
+    emit(Stmt::assume(Expr::lit(false)));
```

Comment:
Why do we need this assumption? Please clarify. Thx!

Reply:
I realized that this is what will happen anyway. Panics are always terminal.